src/App.js

```diff
@@ -1,85 +1,257 @@
 import React from 'react';
+import TodoApp from './Component/TodoApp/TodoApp';
+import TodoList from './Component/TodoList/TodoList';
+import Nav from './Component/Nav/Nav';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  state = {
+    activePage: 'All',
+    id: 0,
+    ListofTodo: [],
+  }
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
+  componentDidMount() {
+    if (localStorage.getItem('list') !== null) {
+      const s = JSON.parse(localStorage.getItem('list'));
+
+      this.setState(s);
+    }
+  }
+
+  componentDidUpdate = () => {
+    localStorage.setItem('list', JSON.stringify(this.state));
+  }
+
+  handleSubmit = (event, newTodoText) => {
+    event.preventDefault();
+
+    if (newTodoText.replace(/ /g, '') === '') {
+      return;
+    }
+
+    const sample = {
+      id: this.state.id + 1,
+      text: newTodoText.trim(),
+      elementState: {
+        completed: false,
+        editing: false,
+      },
+    };
+
+    this.setState(state => ({
+      ListofTodo: state.ListofTodo.concat([{ ...sample }]),
```

Comment:
```suggestion
      listofTodo: [...state.listOfTodo, sample],
```
there is a less bulky way to add a new element